numba/npyufunc/array_exprs.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import print_function, division, absolute_import
+import ast
+
+from numpy import ufunc
+
+from .. import ir, types, rewrites, six, config
+from ..typing import npydecl
+from ..targets import npyimpl
+from .dufunc import DUFunc
+
+
+@rewrites.register_rewrite
```

Comment:
I was going for some symmetry with typing and lowering, but this is a good point that I considered while writing this.  My current opinion is that rewrites should be composed into a singular class if they are going to have any interactions.  Otherwise, they should not interact with each other.  Another possibility is adding optional composition rules to the decorator, when needed.